ListingManager/ListingInformation.cs

```diff
@@ -28,11 +33,17 @@ public ListingInformation(string listingPath)
                 ListingSuffix = !string.IsNullOrWhiteSpace(matches.Groups[3].Value) ? matches.Groups[3].Value : "";
                 ListingDescription = !string.IsNullOrWhiteSpace(matches.Groups[5].Value) ? matches.Groups[5].Value : "";
                 TemporaryPath = listingPath + TemporaryExtension;
+                ListingExtension = System.IO.Path.GetExtension(listingPath);
             }
             else
             {
                 throw new ArgumentException("Listing information not successfully able to be parsed from listing path.", nameof(listingPath));
             }
-        }
+        }
+
+        [GeneratedRegex("Listing(\\d{2}).(\\d{2})([A-Za-z]*)(\\.{1}(.*))?[.cs,.xml]$")]
+        private static partial Regex ExtractListingNameFromAnyApprovedFileTypes();
+        [GeneratedRegex("Listing(\\d{2}).(\\d{2})([A-Za-z]*)(\\.{1}(.*))?.cs$")]
+        private static partial Regex ExtractListingNameFromCSFile();

```

Comment:
Maybe I am missing something, but why do we need both? These look to be the exact same with just the addition of the xml extension. The real program always seems to pass false. So don't we just want the false case and update the tests?

Reply:
Removed Only cs file case